instrumentation/azure-functions/src/main/java/io/opentelemetry/instrumentation/auto/azure/functions/AzureFunctionsInstrumentation.java

```diff
@@ -86,5 +91,25 @@ public static Scope methodEnter(@Advice.Argument(0) final Object request)
     public static void methodExit(@Advice.Enter final Scope scope) {
       scope.close();
     }
+
+    /**
+     * Azure Function Linux with consumption plan doesn't come with 'APPLICATIONINSIGHTS_CONNECTION_STRING' env variable during the initialization.
+     * Need to set this connection string lazily when it becomes available.
+     */
+    private static void lazilySetConnectionString() {
+      if (!AiConnectionString.hasConnectionString()) {
+        String connectionString = System.getenv("APPLICATIONINSIGHTS_CONNECTION_STRING");
+        if (connectionString != null && !connectionString.isEmpty()) {
+          AiConnectionString.setConnectionString(connectionString);
+          // TODO to be deleted later once the testing is completed
+          log.debug("Lazily set the connection string for Azure Function Linux Consumption Plan" + connectionString);
+        } else {
```

Comment:
fall back to `APPINSIGHTS_INSTRUMENTATIONKEY` to support azure functions that were created prior to `APPLICATIONINSIGHTS_CONNECTION_STRING`